crates/noirc_evaluator/src/ssa_refactor/ir/basic_block.rs

```diff
@@ -52,14 +52,23 @@ impl BasicBlock {
         self.instructions.push(instruction);
     }
 
+    /// Retrieve a reference to all instructions in this block.
     pub(crate) fn instructions(&self) -> &[InstructionId] {
         &self.instructions
     }
 
+    /// Sets the terminator instruction of this block.
+    ///
+    /// A properly-constructed block will always terminate with a TerminatorInstruction -
+    /// which either jumps to another block or returns from the current function. A block
+    /// will only have no terminator if it is still under construction.
     pub(crate) fn set_terminator(&mut self, terminator: TerminatorInstruction) {
         self.terminator = Some(terminator);
     }
 
+    /// Returns the terminator of this block.
+    ///
+    /// If this block is not still under construction, this is expected to always be Some.
```

Comment:
Does this read clearer and or cause confusion "If this block has been finalized, this is expected to be Some` ?

Reply:
Changed